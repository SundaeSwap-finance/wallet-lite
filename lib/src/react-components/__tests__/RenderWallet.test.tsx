import { render } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import { describe, expect, it } from "bun:test";

import { IAssetAmountMetadata } from "@sundaeswap/asset";
import {
  RenderWallet,
  TUseWalletObserverState,
  WalletObserverProvider,
} from "../../index.js";

const TestComponent = (
  state: Omit<
    TUseWalletObserverState<IAssetAmountMetadata>,
    "connectingWallet" | "syncingWallet" | "ready"
  >,
) => {
  return (
    <pre>
      {state.activeWallet?.toString()}
      {(state.adaBalance.amount > 0n).toString()}
      {state.isCip45?.toString()}
      {state.mainAddress?.toString()}
      {state.network?.toString()}
      {state.unusedAddresses.toString()}
      {state.usedAddresses.toString()}
      <button
        data-testid="connect"
        onClick={() => state.observer.connectWallet("eternl")}
      ></button>
    </pre>
  );
};

describe("RenderWallet", () => {
  it("should call useWalletObserver", async () => {
    const user = userEvent.setup();

    const { container, getByTestId, rerender } = render(
      <RenderWallet render={TestComponent} />,
      {
        wrapper(props) {
          return <WalletObserverProvider {...props} />;
        },
      },
    );

    expect(container.innerHTML).toMatchSnapshot();

    const button = getByTestId("connect");
    await user.click(button);

    rerender(<RenderWallet render={TestComponent} />);

    expect(container.innerHTML).toMatchSnapshot();
  });
});
