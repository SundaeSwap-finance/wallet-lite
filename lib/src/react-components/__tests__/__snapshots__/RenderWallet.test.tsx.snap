// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderWallet should call useWalletObserver 1`] = `"<pre>falsefalse<button data-testid="connect"></button></pre>"`;

exports[`RenderWallet should call useWalletObserver 2`] = `"<pre>falsefalse<button data-testid="connect"></button></pre>"`;
